emulator/src/ppu/bg_state.rs

```diff
@@ -10,11 +10,12 @@ pub(super) struct TileBitPlanes([u8; 2]);
 
 impl TileBitPlanes {
     pub(super) fn to_u16(self) -> u16 {
-        unsafe { std::mem::transmute(self) }
+        (self.0[0] as u16) << 8 | self.0[1] as u16
```

Comment:
This is actually incorrect, since the two bytes in ‘TileBitPlanes’ are stored in little-endian format! This means that ‘self.0[0]’ should actually become the low byte in the u16, while ‘self.0[1]’ should become the high byte. So the correct code should shift out ‘self.0[1]’ and or in ‘self.0[0]’. Though really, I’d recommend just using the ‘u16::from_le_bytes()’ function provided by std to do this instead.

Reply:
Oh thanks, I didn't even realize that was a function in the std library. Fixed in efea267